### PR TITLE
Lowercase all words in VCF

### PR DESCRIPTION
**Pull request by torbjorgen**
This PR will make all data in VCF to be lower-case when fed as input to ML model,
in the text pre-processing step.

The main reason for doing this is to make the vocabulary case-insensitive
(smaller vocabulary) as well as avoiding the scenario when
variant meta data is multi-case;

`Pathogenic` vs `pathogenic`

I don't want these two words to be separately encoded in the embeddings,
risk of model overfitting to upper-case words and ignoring variants
in lower-case, and vice versa.

AFAIK I don't see any risk (mixups, data-loss, .. ?) in doing this.
Feel free to let me know if I'm missing anything here.

### CODE REVIEW

File: src/rdds/lib/tf/text_preprocessing_layer.py [hunk 1]
```diff
@@ -19,15 +19,20 @@ def __init__(self,
         tf.Module.__init__(self, name='TextPreprocessing')
         self._splitter: tftext.Splitter = tftext.RegexSplitter(split_regex=split_regex)
 
+    def _process_tensor(self, tensor):
```
Comment (henrikstranneheim) [inline src/rdds/lib/tf/text_preprocessing_layer.py]: Add type hint and return hint

File: src/rdds/lib/tf/text_preprocessing_layer.py [hunk 2]
```diff
@@ -19,15 +19,20 @@ def __init__(self,
...
+        tensor = tf.strings.lower(tensor)  # lowercase all words
+        tensor = self._splitter.split(tensor)  # split words
+        return tensor
```
Comment (henrikstranneheim) [inline src/rdds/lib/tf/text_preprocessing_layer.py]:
```suggestion
        return self._splitter.split(tensor)
```

File: src/rdds/lib/tf/text_preprocessing_layer.py [hunk 3]
```diff
@@ -19,15 +19,20 @@ def __init__(self,
         tf.Module.__init__(self, name='TextPreprocessing')
         self._splitter: tftext.Splitter = tftext.RegexSplitter(split_regex=split_regex)
 
+    def _process_tensor(self, tensor):
+        tensor = tf.strings.lower(tensor)  # lowercase all words
```
Comment (henrikstranneheim) [inline src/rdds/lib/tf/text_preprocessing_layer.py]:
```suggestion
        tensor = tf.strings.lower(tensor)
```

**Review by henrikstranneheim — APPROVED** [review]
👍

**Comment by torbjorgen** [comment]
Adjusted according to comments by Henrik.

**Pull request merged**